### DISCUSSION TITLE
How to store LFS in OneDrive and manage from github ?

### DISCUSSION THREAD

**Discussion by douglassoarez** [git-lfs/git-lfs#4724] [Q&A]
I need to work with solidworks file versioning collaboratively, github would work perfectly for that, but all files would take up +10GB easily. How can I use LFS to store in OneDrive and manage via github ?

**Comment by bk2204** [comment]
It is technically possible to set the `lfs.storage` config value for your repository to a local location (on Windows, that path should use forward slashes).  So it would look like `git config lfs.storage /mnt/onedrive/lfs/project`.  You could also create a Git repository on OneDrive and then push the data there by using a `file:///` remote.

However, we know that cloud syncing services like OneDrive and Dropbox do corrupt Git repositories, and they may well corrupt LFS storage, so while you can do this, we don't recommend it since we don't want you to experience data loss.

**Reply by ok-martin** [reply]
Hey, which do you mean will corrupt a git repository: 
1. using LFS with a folder synced by OneDrive/Dropbox to store and collaborate large files in a project or 
2. having a git repository on OneDrive/Dropbox
3. both

The second definitely makes sense but I thought github and LFS was ok to work with the first option.

**Reply by bk2204** [reply]
The answer is probably 3.  We've definitely seen cases where having a Git repository on a cloud syncing service will cause corruption, but I haven't seen a specific case of corruption with `lfs.storage`, although I suspect that such a problem is likely.  I'd recommend not doing either.

**Reply by ok-martin** [reply]
Thanks. Is it the same if I use OneDrive/Dropbox etc through the Custom Transfer Agent? https://github.com/git-lfs/git-lfs/blob/main/docs/custom-transfers.md

**Reply by bk2204** [reply]
Yes, my response is the same no matter how you store data in a cloud syncing service.  The problem is that they work file by file and not as a whole unit like a directory, which means that they can end up having partial data.
